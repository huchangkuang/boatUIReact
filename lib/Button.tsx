import React, {FC} from 'react';

type Props = {

};
const Button: FC<Props> = (props: Props) => {
  return (
    <div>
      Button
    </div>
  );
};
export default Button