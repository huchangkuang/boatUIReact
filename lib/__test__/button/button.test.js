import 'jsdom-global/register';
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from "enzyme";
import Button from "../../button/button";

configure({adapter: new Adapter()});
describe('icon', () => {
  it('should be div', function () {
    const component = renderer.create(<Button name='share' />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('can click', function () {
    const fn = jest.fn()
    const wrapper = shallow(<Button name='share' onClick={fn} />);
    const icon = wrapper.find('.boat-button')
    icon.simulate('click')
    expect(fn).toBeCalled()
  });
})