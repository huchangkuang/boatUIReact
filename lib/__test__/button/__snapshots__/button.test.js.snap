// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon should be div 1`] = `
<div
  className="boat-button"
  name="share"
/>
`;
